src/components/tl/ucp/tl_ucp.h

```diff
@@ -23,7 +24,7 @@ typedef struct ucc_tl_ucp_lib_config {
 
 typedef struct ucc_tl_ucp_context_config {
     ucc_tl_context_config_t super;
-    int                     test_param;
+    int                     preconnect;
```

Comment:
I'm not sure what is the role of pre-connect here: flag or storing the threshold value? If it is the threshold value, I think preconnect should be part of the team rather than the context. The context can be used for any team size, and if your intent to wire based on the team size, then it makes sense that it is a part of the team structure.

Reply:
Also, do you want it is as uint32/64 - similar to team size?

Reply:
I've update the PR. Preconnect is now a threshold. Threshold that defines the number of ranks in the UCC team/context below which the team/context enpoints will be preconnected during  corresponding team/context create call. We can only store the parameter on the  context. Firstly we don't have interface to set parameters per team. And since user anyways can only set one parameter value for a job then it is constant for all teams anyways. The actual behavior will depend on other parameters and settings. Basically, if UCP endpoints are stored on a team then preconnect applies to the team. If eps are stored on a context then the threshold will define if the whole tl ucp context gets preconnected